addon/connection.cc

```diff
@@ -106,18 +106,23 @@ Napi::Value Connection::Execute(const Napi::CallbackInfo &info) {
 }
 
 Napi::Value Connection::Close(const Napi::CallbackInfo &info) {
-  for (auto &result : *results) {
-    if (result != nullptr) {
-      result->close();
-    }
+  // the following gives segfaults for some reason now
+  // for (auto &result : *results) {
+  //   if (result != nullptr) {
+  //     result->close();
+  //   }
+  // }
+  if (connection) {
+    connection.reset();
+  }
+  if (database) {
+    database.reset();
   }
-  connection.reset();
-  database.reset();
   return info.Env().Undefined();
 }
 Napi::Value Connection::IsClosed(const Napi::CallbackInfo &info) {
   Napi::Env env = info.Env();
-  bool isClosed = connection == nullptr || connection->context->is_invalidated;
+  bool isClosed = connection == nullptr;
```

Comment:
Is this even accurate? If DuckDB doesn't provide this, should we?

Reply:
This is for when we want to destroy the underlying connection/database without waiting for GC to call the destructors. I tested it initially (when implementing the close logic originally) and what I have found is that you can have a lot of data in the address space of the process but outside of the V8 heap and so GC will not get triggered (i.e. DuckDB loads something into memory taking 20 GB, but obviously DuckDB is outside of the V8 heap so GC is not triggered. Yes, actually triggering the GC will free that memory, but it won't be automatic based on the size). You can have a process taking 20GB of memory even though the max limit for V8 heap is 1.7 GB. And when I called `close` that memory was freed. So, I think we still need it.